src/Toolkit/A.php

```diff
@@ -47,6 +47,17 @@ public static function apply(array $array, ...$args): array
 		return $array;
 	}
 
+	/**
+	 * Counts the number of elements in an array
+	 *
+	 * @param array $array
+	 * @return int
+	 */
+	public static function count(array $array = []): int
```

Comment:
Just saw that this argument is optional. Why is that?

Reply:
You know, I don't think there is a great need for that - I think it's just automatic defensive default value on my end.

Reply:
I'd be totally fine with it. I was just confused because we don't do it for any other `A` method. We do it in the `Str` class in a few places and I always found that weird. After all the caller wants the method to do something, so why call it without arguments?

Reply:
I think we have it in some places for when you pass a field or method call directly to these method that could also yield `null` - that way the method doesn't crash into an error but gracefully still gets you, e.g. `0` in this case. Which is probably a valid result in most cases when you wanted to count something that got you `null`.

Reply:
That's not what this does though. You cannot pass `null`, the default just allows you to call the method without any argument at all.

Reply:
True I think what I described we do in the Str class. Either it would need to be `= null` here or not optional.

Reply:
I think it should be not optional for consistency with all the other `A` methods.

Reply:
👍